### ISSUE TITLE
Task: Create a triplestore for DAAN data for the Media Suite

### ISSUE THREAD

**Issue by mwigham** [beeldengeluid/beng-lod-server#116]
At present we have a triplestore of iMMix data that we use for the Media Suite, both for linked data for the iMMix index and for the DAAN index. We need a triplestore of DAAN data so that we have up-to-date data and we avoid issues due to mismatches in the data, such as [this one](https://github.com/beeldengeluid/x-omgeving/issues/396).

Tasks:
- [ ] Decide if we need to temporarily have two triplestores or if we can simply replace the current one (the iMMix index in the Media Suite is going to be removed at some point
- [ ] Run the importer to get all the triplestore data
- [ ] Load the data into the triplestore

Acceptance criteria:
- [ ] The data for the DAAN index is available as linked data in the triplestore

**Comment by wmelder** [comment]
@mwigham there is one already here: http://zorin.beeldengeluid.nl:3023/browse/list_graphs
Will that work for you? It has the nsiv data. I was planning to also load the sdo data in a named graph.

**Comment by mwigham** [comment]
wow, fantastic! Is it ok to use that one for production?

**Comment by wmelder** [comment]
Well, I was sort of thinking we could use this for testing. Now that we are producing dumps in several formats we might also want to play around with the data. Just to see if it is what we expected.

Is it ok if you just use it now and once you need something for production, we setup a new one? 

By the way, this one only contains the RDF for the programs, as we get from the beng-lod server.

**Comment by mwigham** [comment]
Ok, I understand. This is fine for testing, and indeed handy to play around a bit. 
Then we keep this issue open until we have a production version ready.

**Comment by wmelder** [comment]
for production you might also want a more dynamic updating process in place? Currently, the dump was built on top of the ids from the aggregated index in the media suite. It would already be better if we incorporate all resources (season, series, scene description), but also keep it up to date (delete stuff etc.) using the flex datastore pusher and SPARQL update. But we might also go for weekly dumps, or something like that.

**Comment by mwigham** [comment]
Yes, we'll definitely want a good update process too, but that is then a separate issue for later. Having a DAAN-based store for production, even though it's not regularly updated, is already a good step in itself.

**Comment by jblom** [comment]
I'd say in general we don't have a great production server and automated deployment workflow yet, but in practice we do use zorin for the MS production, so another triplestore on that server qualifies for now.

**Comment by wmelder** [comment]
@mwigham 
The triple store listed above currently has data for all catalogue types (incl. scene_descriptions) in SDO format. This can be updated periodically or when some additional data feature becomes available. When it needs to get a MediaSuite production status, it means I need to spawn a new container for testing. Thus, we might also prepare a dedicated MediaSuite production triple store. But what data does that need? Does it also need the GTAA personnames ? and the alignment wikidata entities?  If so, we might consider to just additionally load the DAAN SDO data in the current [Person Graph triple store](http://zorin.beeldengeluid.nl:3024/browse/list_graphs). Or is there an explicit need to keep these graphs separate? It would simplify querying if not two but one triple store needs to be queried...

**Comment by mwigham** [comment]
For production we need the DAAN data and the person graph triple store. We had kept them separate to keep the person graph lightweight. The queries are currently specific per graph so there is no effect on querying.

**Comment by wmelder** [comment]
Does the Media Suite need the descriptions and summaries as well? Or, do we need the download the data in the NISV profile or SDO profile?

**Comment by mwigham** [comment]
At present we only use the program links with persons and their roles. The query works with the nisv profile, but it should be easy enough to convert it to sdo if that's better.

**Comment by jblom** [comment]
So far we don't really have a tool or functionality for using the DAAN LOD (we only require the entity graphs for autocomplete & entity search, etc). Until we e.g. start rebuilding an LD explore tool, we should be fine without the descriptions & summaries?

**Comment by mwigham** [comment]
Yes. But don't forget that it's not just the person graph, we also need the links between programmes and persons. Otherwise the nice list in the Resource Viewer disappears ;)

Just for completeness, the autocomplete doesn't use the LOD. It uses our copy of the GTAA in the Elastic Search cluster. In the future we will possibly switch to the new thesaurus layer the A team is working on.

**Comment by jblom** [comment]
check @mwigham ah yes the links, which could/should be stored in a separate graph as well? (e.g. graph1 = DAAN; graph2 = person data; graph3=links between the two) In this case you would need to query across graphs.

I guess now you store the links to DAAN in the person graph? (which is fine for now as well, just curious about how you set it up right now)

**Comment by mwigham** [comment]
At present we have one iMMix graph (yes, iMMix still!) with the programme
metadata, and one with the person information. Both are hosted on Zorin. We
only use two queries:
- retrieve all information about a person from the person graph (for the
person pop-up, does NOT include information about the programmes they
appear in)
- retrieve the persons linked to a programme (or its series/season/scene
descriptions) along with their role in the programme from the iMMix graph
(for the Resource Viewer)

Querying across graphs is therefore not necessary at present. We would need
this later if we want to use person details in combination with programme
information e.g. search for all programmes containing persons belonging to
a certain political party.

Op wo 29 sep. 2021 om 09:09 schreef Jaap Blom ***@***.***>:

**Comment by jblom** [comment]
@mwigham thanks for the overview. I  am really for a WM, MW, JB whiteboard session. No rush, but I got the itch :p

**Comment by mwigham** [comment]
I'm in :)

**Comment by wmelder** [comment]
Currently, the Media Suite then probably uses [this store ](http://zorin.beeldengeluid.nl:3020/home) for the iMMix linked data. [This server](http://zorin.beeldengeluid.nl:3023/home) contains the DAAN SDO linked data and is used by the example UI. It should contain the links between persons and programs and roles. We can make this a dedicated MediaSuite production server, if you all agree. 

At the moment we are deploying the SDO data in AWS, so the example UI can then also use that version (publicly available). When I need to test with loading data etc. I will spawn a new container. Please let me know if MediaSuite no longer needs the iMMix data, because then I will shut the server on port 3020 down.

For completeness sake [here ](http://zorin.beeldengeluid.nl:3024/home) is the server for the person graph.

In summary, on zorin we have several servers running on different ports:

```http://zorin.beeldengeluid.nl:<port>```
3020: iMMix NISV
3023: DAAN SDO
3024: Person graph

Coming soon:
AWS: DAAN SDO
